activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -638,8 +638,15 @@ def find_last(limit)
       end
 
       def ordered_relation
-        if order_values.empty? && !_order_columns.empty?
-          order(_order_columns.map { |column| table[column].asc })
+        if order_values.empty?
+          if !_order_columns.empty?
+            return order(_order_columns.map { |column| table[column].asc })
+          end
+
+          raise OrderError, <<~MSG.squish
+            Relation has no current order, and #{model} has no order columns to use as a default.
+            At least one of `implicit_order_column`, `query_constraints`, or `primary_key` must be set on the model.
```

Comment:
I think `must be set` isn't quite correct, because you could also specify an `order` on the relation? (same concern with the change to the other error)

Reply:
> because you could also specify an order on the relation?

Yes that's what this first bit is alluding to:

> Relation has no current order,

But I can make it more clear by specifying that in the suggestion as well 👍🏽

Reply:
Done, let me know if it's clearer now.